app/notifications/validators.py

```diff
@@ -147,17 +148,28 @@ def check_notification_content_is_not_empty(template_with_content):
         raise BadRequestError(message=message)
 
 
-def validate_template(template_id, personalisation, service, notification_type):
+def get_template_dict(template_id, service_id):
```

Comment:
We can add a long-lived cache here, which can be shared with, and invalidated by the admin app
```suggestion
@request_cache.set('template-{template_id}-version-None')
def get_template_dict(template_id, service_id):
```

Reply:
I'd be tempted to roll the cache change as a separate deploy so we can compare performance, thoughts?

Reply:
Agree, this was just to show what that change would look like, and why it makes sense to split the code up into these specific functions.